src/i_video.c

```diff
@@ -725,6 +725,35 @@ static void I_ResetTargetRefresh(void);
  #define I_CpuPause()
 #endif
 
+static void I_WaitUntil(uint64_t target_time)
+{
+    while (true)
+    {
+        uint64_t current_time = I_GetTimeUS();
+        uint64_t elapsed_time = current_time - frametime_start;
+        uint64_t remaining_time = 0;
+
+        I_CpuPause();
+
+        if (elapsed_time >= target_time)
+        {
+            frametime_start = current_time;
+            break;
+        }
+
+        remaining_time = target_time - elapsed_time;
+
+        if (remaining_time > 1000)
+        {
+            I_SleepUS(500);
+        }
+        else
+        {
+            I_Sleep(0); // yield
```

Comment:
Do we need `I_Sleep(0)`? On Windows, SDL will call [`Sleep(0)`](https://learn.microsoft.com/en-us/windows/win32/api/synchapi/nf-synchapi-sleep#parameters) witch makes some sense, but I think this is not portable.

Reply:
Yes, on Windows it reduces CPU usage for me and gives a nice latency improvement (which directly yielding the thread via e.g. `SwitchToThread` does not provide). Do you know if there is a posix counterpart to `Sleep(0)`?

At least this should be no worse than the existing implementation, which has many calls to `Sleep(0)` (when remaining time < 2ms).

Reply:
I think there is a `pthread_yield`, but I'd rather avoid that complication. How about `I_SleepUS(100)` instead of `I_Sleep(0)` or something similar?

Reply:
It's too coarse, in my tests `I_SleepUS(100)` often sleeps for 1 ms whereas `I_Sleep(0)` sleeps only 10-30us or so.

Reply:
Okay, maybe it makes sense to add the `I_ThreadYield` function, with POSIX it's [`sched_yield`](`https://www.man7.org/linux/man-pages/man2/sched_yield.2.html`), on Windows it's just `Sleep(0)`. How do you test this?

Reply:
> try linux and let me know if I broke anything

I only ran a quick test with `mangohud --dlsym woof` but the frametimes were identical between the master branch and this PR. OpenGL render driver, 9900K, RTX3080, Arch.